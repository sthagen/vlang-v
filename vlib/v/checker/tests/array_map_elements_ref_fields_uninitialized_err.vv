struct Foo {
	n &int
}

type AliasFoo = Foo

fn main() {
	_ = []Foo{}
	_ = [1]Foo{}
	_ = map[string]Foo{}
	_ = map[string][]Foo{}
	_ = []AliasFoo{}
}
