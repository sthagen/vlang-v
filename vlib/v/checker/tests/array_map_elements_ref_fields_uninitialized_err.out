vlib/v/checker/tests/array_map_elements_ref_fields_uninitialized_err.vv:8:6: notice: reference field `Foo.n` must be initialized (part of struct `Foo`)
    6 | 
    7 | fn main() {
    8 |     _ = []Foo{}
      |         ~~~~~~
    9 |     _ = [1]Foo{}
   10 |     _ = map[string]Foo{}
vlib/v/checker/tests/array_map_elements_ref_fields_uninitialized_err.vv:9:6: notice: reference field `Foo.n` must be initialized (part of struct `Foo`)
    7 | fn main() {
    8 |     _ = []Foo{}
    9 |     _ = [1]Foo{}
      |         ~~~~~~~~
   10 |     _ = map[string]Foo{}
   11 |     _ = map[string][]Foo{}
vlib/v/checker/tests/array_map_elements_ref_fields_uninitialized_err.vv:10:6: notice: reference field `Foo.n` must be initialized (part of struct `Foo`)
    8 |     _ = []Foo{}
    9 |     _ = [1]Foo{}
   10 |     _ = map[string]Foo{}
      |         ~~~~~~~~~~~~~~~~
   11 |     _ = map[string][]Foo{}
   12 |     _ = []AliasFoo{}
vlib/v/checker/tests/array_map_elements_ref_fields_uninitialized_err.vv:11:6: notice: reference field `Foo.n` must be initialized (part of struct `Foo`)
    9 |     _ = [1]Foo{}
   10 |     _ = map[string]Foo{}
   11 |     _ = map[string][]Foo{}
      |         ~~~~~~~~~~~~~~~~~~
   12 |     _ = []AliasFoo{}
   13 | }
vlib/v/checker/tests/array_map_elements_ref_fields_uninitialized_err.vv:12:6: notice: reference field `Foo.n` must be initialized (part of struct `Foo`)
   10 |     _ = map[string]Foo{}
   11 |     _ = map[string][]Foo{}
   12 |     _ = []AliasFoo{}
      |         ~~~~~~~~~~~
   13 | }
